File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/telemetry/TextFormat.java
package org.firstinspires.ftc.teamcode.util.telemetry;

public class TextFormat {

    // --- Base HTML Wrappers ---
    public static String color(String text, String hexColor) {
        return "<font color='" + hexColor + "'>" + text + "</font>";
    }

    public static String bold(String text) {
        return "<strong>" + text + "</strong>";
    }

    public static String italic(String text) {
        return "<em>" + text + "</em>";
    }

    public static String big(String text) {
        return "<big>" + text + "</big>";
    }

    public static String small(String text) {
        return "<small>" + text + "</small>";
    }

    // --- Semantic Presets ---
    public static String header(String text) {
        return big(bold(color(text, "#448aff"))); // FTC Blue
    }

    public static String subheader(String text) {
        return bold(color(text, "#448aff")); // FTC Blue
    }

    public static String success(String text) {
        return bold(color(text, "#00ff88")); // Bright Green
    }

    public static String error(String text) {
        return bold(color(text, "#ef5350")); // Alert Red
    }

    public static String warn(String text) {
        return color(text, "#ffeb3b"); // Yellow
    }

    // --- Utilities ---

    /**
     * Strips all HTML tags from a string. Required for Panels compatibility.
     */
    public static String stripHtml(String html) {
        if (html == null) return null;
        return html.replaceAll("<[^>]*>", "");
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/kotlin_mirror/util/telemetry/TextFormat.kt
package org.firstinspires.ftc.teamcode.kotlin_mirror.util.telemetry

/**
 * Utility class for formatting text with HTML tags for the Driver Station telemetry.
 *
 * Provides methods for coloring, sizing, and styling text, as well as semantic presets
 * for headers, success messages, errors, and warnings.
 */
object TextFormat {

    // --- Base HTML Wrappers ---

    @JvmStatic
    fun color(text: String, hexColor: String): String {
        return "<font color='$hexColor'>$text</font>"
    }

    @JvmStatic
    fun bold(text: String): String {
        return "<strong>$text</strong>"
    }

    @JvmStatic
    fun italic(text: String): String {
        return "<em>$text</em>"
    }

    @JvmStatic
    fun big(text: String): String {
        return "<big>$text</big>"
    }

    @JvmStatic
    fun small(text: String): String {
        return "<small>$text</small>"
    }

    // --- Semantic Presets ---

    @JvmStatic
    fun header(text: String): String {
        return big(bold(color(text, "#448aff"))) // FTC Blue
    }

    @JvmStatic
    fun subheader(text: String): String {
        return bold(color(text, "#448aff")) // FTC Blue
    }

    @JvmStatic
    fun success(text: String): String {
        return bold(color(text, "#00ff88")) // Bright Green
    }

    @JvmStatic
    fun error(text: String): String {
        return bold(color(text, "#ef5350")) // Alert Red
    }

    @JvmStatic
    fun warn(text: String): String {
        return color(text, "#ffeb3b") // Yellow
    }

    // --- Utilities ---

    /**
     * Strips all HTML tags from a string.
     *
     * Required for compatibility with telemetry sinks that do not support HTML,
     * such as the FTControl Panels log.
     *
     * @param html The HTML string to strip.
     * @return The plain text string.
     */
    @JvmStatic
    fun stripHtml(html: String?): String? {
        return html?.replace(Regex("<[^>]*>"), "")
    }
}
